### ISSUE TITLE
Ventura 13.0 - Not triggering screen saver

### ISSUE THREAD

**Issue by sdugoten** [JohnCoates/Aerial#1271]
#### Required information
In order to help us sort your issue, we ask that you provide the following information:
- [ ] Mac model: Mac Studio M1 Max
- [ ] macOS version:  Ventura 13.0
- [ ] Monitor setup: Studio Display
- [ ] Aerial ver: 1.2.2
 
#### Description of issue / Feature request

 I set hot corner on upper right to trigger the screen saver.   Lower right to trigger the sleep.

After the Mac is sleep for like 5 to 10 mins.  If you wake up the Mac, going to the hot corner would no longer trigger the screen saver.  If you let the time pass by, the Mac will automatically go into sleep when it's the time to trigger the screen saver.

The only work around is to change the screen saver to some other screen saver comes with Mac.  Test the hot corner.  And then switch it back to Aerial.    100% reproducible.

**Comment by glouel** [comment]
Hey @sdugoten 

Sorry, I'm not sure I totally understand what you are saying here.

Are you saying that you first trigger sleep via a hot corner, wake it up, then launch screen saver via other hot corner ? 

It's macOS that decides to start the screensaver, so I have a hard time following what's happening here. I do regularly use "Suspend activity" in macOS here in Ventura which should be similiar to your hot corner. When waking up from that, I still do get the screen saver to start as expected via hot corner (or time). 

Please try to give me a bit more info because I'm puzzled here ! Thanks !

**Comment by sdugoten** [comment]
https://streamable.com/apfpi8

Please see the video above.  Baiscally, the video was taken after I put the Mac into sleep for like 5 minutes via the hot corner.  You can see that in the video, hot corner to make it to trigger the screen saver do not work.

**Comment by sdugoten** [comment]
@glouel 

You just need to set two hot corner like what I did in the video.  Then...

1.  Use lower right corner to put the Mac into sleep
2.  Wait for 5 to 10 mins and let the Mac deep sleep
3. Wake up the Mac via a mouse click on space bar
4. Move your mouse to upper right corner to the upper right try to trigger the screen saver
5. You will see the hot corner to trigger the screen saver do not work.

**Comment by sdugoten** [comment]
@glouel 

After step 5 above if you find that hot corner no longer tigger the screen saver, you will find that the screen saver will never trigger at all.  Let say you set the Mac to trigger screen saver after idling for 10 mins.  You leave the Mac idle for 10 mins, the screen saver will never get trigger even after you idle the Mac for 10 mins.

The screen saver is now in a broken state.  There is only 2 way I know how to fix the broken state.

1.  Reboot the Mac

or

2. Kill the Aerial companion app on the top bar.  Manually change the screen saver to something else.  Check if you can trigger the screen saver once again via the hot corner.  IF that works, change the screen saver back to Aerial.

It will run fine until the next time the Mac is deep sleep.  And you will have to do these steps to fix the broken state again.

**Comment by sdugoten** [comment]
I think I know how to reproduce the issue.  the screen saver won't trigger when Barrier is running in Client mode connecting to a Windows barrier server.

https://github.com/debauchee/barrier

Have the Mac run the Barrier client mode
Put the Mac into sleep for 5 mins
Wake up the Mac and use hot corner to trigger the screen saver
It won't do anything even if you try to trigger the screen saver

I don't know what conflict between the Barrier client mode and Aerial .  Stock screensaver do not have this issue.

**Comment by glouel** [comment]
Good to hear, thanks for the update.
